specification/columns/resourcename.md

```diff
@@ -2,7 +2,7 @@
 
 The Resource Name is a display name assigned to a [*resource*](#glossary:resource). It is commonly used for cost analysis, reporting, and allocation scenarios.
 
-The ResourceName column MUST be present in a FOCUS dataset when the provider supports billing based on provisioned resources. This column MUST be of type String. The ResourceName value MAY be a nullable column as some cost data [*rows*](#glossary:row) may not be associated with a *resource* or because a display name cannot be assigned to a *resource*. ResourceName MUST NOT be null if a display name can be assigned to a *resource*. *Resources* not provisioned interactively or only have a system-generated [ResourceId](#resourceid) MUST NOT duplicate the same value as the ResourceName.
+The ResourceName column MUST be present in a FOCUS dataset when the provider supports billing based on provisioned resources. This column MUST be of type String, MUST NOT be null if [ResourceId](#resourceid) is not null and a display name can be assigned to a *resource*, and MUST be null when [CommitmentDiscountStatus](#commitmentdiscountstatus) is "Unused". The ResourceName value MAY be a nullable column as some cost data [*rows*](#glossary:row) may not be associated with a *resource* or because a display name cannot be assigned to a *resource*. *Resources* not provisioned interactively or only have a system-generated ResourceId MUST NOT duplicate the same value as the ResourceName.
```

Comment:
Same issues as resourceID

Reply:
I don't know of a case where a resource name exists and a corresponding resource id does not, but I'm going to back this change out for now to focus more on the Commitment columns.